client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -402,6 +416,105 @@ private void checkIfBlocksFailed() {
     }
   }
 
+  private boolean needReAssignShuffleServer(Set<Long> failedBlockIds) {
```

Comment:
I think we should check the blocks existence that meet the requirements, current impl looks wrong.

Reply:
I got it. I will modify it. thinks.